library/test/src/test/java/com/bumptech/glide/load/resource/gif/GifDrawableTest.java

```diff
@@ -568,12 +569,53 @@ public void testSetAlphaSetsAlphaOnPaint() {
   public void testSetColorFilterSetsColorFilterOnPaint() {
     ColorFilter colorFilter = new PorterDuffColorFilter(Color.RED, Mode.ADD);
     drawable.setColorFilter(colorFilter);
+    verify(paint).setColorFilter(eq(colorFilter));
+  }
+
+  @Config(sdk = Build.VERSION_CODES.LOLLIPOP)
+  @Test
+  public void testDrawSetsTintListColorFilterOnPaint() {
+    ColorStateList tint =
+        new ColorStateList(
+            new int[][] {new int[] {android.R.attr.state_pressed}, new int[0]},
+            new int[] {Color.RED, Color.GREEN});
+    drawable.setTintList(tint);
+    drawable.setTintMode(Mode.ADD);
+    when(paint.getColorFilter()).thenReturn(null);
+    drawable.draw(new Canvas());
+
+    // draw() temporary sets tint filter then restore.
+    verify(paint).setColorFilter(eq(new PorterDuffColorFilter(Color.GREEN, Mode.ADD)));
+    verify(paint).setColorFilter(null);
+
+    assertThat(drawable.setState(new int[] {android.R.attr.state_pressed})).isTrue();
+    drawable.draw(new Canvas());
```

Comment:
performing multiple acts in a test implies that there should be two tests, please split this into a separate test that verifies the state is reflected in the filter.

Reply:
Done.